cmd/log-parser/agent.go

```diff
@@ -114,21 +114,36 @@ func unpackAgentLogEntry_v2(le LogEntry) (agent LogEntry, err error) {
 		agent.Source = "vmconsole"
 		return agent, nil
 	}
-
-	pid, err := strconv.Atoi(result["pid"])
-	if err != nil {
-		return LogEntry{}, fmt.Errorf("failed to convert pid")
+	// we were able to unpack agent msg, let's drop previous Data (runtime entries)
+	agent.Data = make(map[string]string)
+
+	for k, v := range result {
+		switch k {
+		case "msg":
+			agent.Msg = v
+		case "level":
+			agent.Level = strings.ToLower(v)
+		case "source":
+			agent.Source = v
+		case "name":
+			agent.Name = v
+		case "pid":
+			pid, err := strconv.Atoi(v)
+			if err != nil {
+				return LogEntry{}, fmt.Errorf("failed to convert pid")
+			}
+			agent.Pid = pid
+		case "container-id", "cid": // better to ensure consistency in the agent, same for exec-id/eid
+			agent.Container = v
+		// NOTE: we do not take the agent's timestamp into account, because there is a ~1sec delay
+		// for the agent's log to get through. Using the agent's timestamp would then make its logs
+		// appear out of order compared to other logs from the guest.
+		// The agent's logs timestamp is still visible for reference in the Data section of the logs.
+		case "ts":
```

Comment:
As it is, the timestamp will not be visible in the Data section (as explained in the comment above).
You need to add "fallthrough" here, or comment/remove the 'case "ts":' line.
I don't think we should totally remove the agent's timestamp though.

Reply:
Right, done